lib/string_segmenter.rb

```diff
@@ -1,2 +1,43 @@
+require 'dictionary'
+
 def segment_string(str)
-end
\ No newline at end of file
+#1. create an array to store words
+	arr = []
+#2. loop over the first character and check if it's a word according to dictionary.rb
+	str.each do |n|
+	#check if character 1 is word
+	if valid_word?(n)
+		#if it is, store it in the last place of the array
+			arr[-1] << n
```

Comment:
Does `arr[-1] << n` work as you expect it to?

Since `arr[any_number]` is meant to return a _specific value_ (e.g. `["Foo", "Bar", "Lorem"][-1]` returns `"Lorem"`), it looks like `arr[-1] << n` is similar to doing `"Lorem" << n` (which will just add the strings together–not add `n` as a new element in the array).